src/eip7594/eip7594.c

```diff
@@ -612,6 +669,47 @@ static C_KZG_RET compute_commitment_to_aggregated_interpolation_poly(
     return ret;
 }
 
+/**
+ * Compute weighted sum of proofs
+ *
+ * @param[out] weighted_proof_lincomb The resulting G1 sum of the proofs scaled by coset factors
+ * @param[in]  proofs_g1              Array of G1 elements representing the proofs
+ * @param[in]  r_powers               Array of powers of r used for weighting
+ * @param[in]  cell_indices           Array of cell indices
+ * @param[in]  num_cells              The number of cells
+ * @param[in]  s                      The trusted setup containing roots of unity
+ */
+static C_KZG_RET computed_weighted_sum_of_proofs(
+    g1_t *weighted_proof_sum_out,
+    const g1_t *proofs_g1,
+    const fr_t *r_powers,
+    const uint64_t *cell_indices,
+    uint64_t num_cells,
+    const KZGSettings *s
+) {
+    C_KZG_RET ret;
+    fr_t coset_factor_pow;
+    fr_t *weighted_powers_of_r = NULL;
+
+    ret = new_fr_array(&weighted_powers_of_r, num_cells);
+    if (ret != C_KZG_OK) goto out;
+
+    for (size_t i = 0; i < num_cells; i++) {
+        uint64_t pos = reverse_bits_limited(CELLS_PER_EXT_BLOB, cell_indices[i]);
+        fr_t coset_factor = s->roots_of_unity[pos];
+        // Compute h_k^n
```

Comment:
I wonder if someone reading this really knows what `h_k` and `n` is.
Maybe you want to say something like 
```suggestion
        // Compute h_k^n, with h_k and n as in the spec.
```